flutter/news/lib/features/frontpage/data/datasource/articles_remote_data_source.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:news/core/constants.dart';
+import 'package:news/core/error/failures.dart';
+import 'package:news/core/result.dart';
+import 'package:news/features/frontpage/data/models/article_model.dart';
+import 'package:news/features/frontpage/data/models/base_news_response_model.dart';
+
+class ArticlesRemoteDataSource {
+  final http.Client client;
+
+  ArticlesRemoteDataSource({required this.client});
+
+  Future<Result<List<ArticleModel>>> topHeadLines() async {
+    var url = Uri.https(Constants.baseUrl, Constants.topHeadlinesEndpoint, {'country': 'us'});
+
+    var response = await client.get(url, headers: {'X-Api-Key': Constants.newsApiKey});
+    if (response.statusCode == 200) {
+      return Result.success(
+          BaseNewsResponseModel.fromJson(json.decode(response.body)).articles
+              as List<ArticleModel>);
+    } else {
+      throw ServerFailure("Server failure");
```

Comment:
🛑 

Here we should be returning a `Result.failure(ServerFailure)`

Reply:
updated on [34d2f99](https://github.com/novoda/multiplatform-playground/pull/15/commits/34d2f99e34fbfc03733e882fdcc758565e92553f)